backend/de.metas.banking/de.metas.banking.base/src/main/java/de/metas/banking/payment/IPaySelectionBL.java

```diff
@@ -1,24 +1,33 @@
 package de.metas.banking.payment;
 
 import com.google.common.collect.ImmutableSet;
+import de.metas.banking.BankAccountId;
 import de.metas.banking.BankStatementAndLineAndRefId;
 import de.metas.banking.BankStatementLineId;
 import de.metas.banking.PaySelectionId;
 import de.metas.banking.PaySelectionLineId;
 import de.metas.bpartner.BPartnerId;
+import de.metas.invoice.InvoiceId;
+import de.metas.money.CurrencyId;
 import de.metas.payment.PaymentId;
 import de.metas.util.ISingletonService;
 import lombok.NonNull;
 import org.compiere.model.I_C_PaySelection;
 import org.compiere.model.I_C_PaySelectionLine;
+import org.jetbrains.annotations.Nullable;
 
 import java.util.Collection;
 import java.util.Map;
 import java.util.Optional;
+import java.util.Properties;
 import java.util.Set;
 
 public interface IPaySelectionBL extends ISingletonService
 {
+	@Nullable BankAccountId getBankAccountId(@NonNull InvoiceId invoiceId,
+											 @NonNull CurrencyId currencyId,
+											 @NonNull Properties ctx);
```

Comment:
consider getting rid of ctx param. try avoiding the usage of `ctx` if possible.
Where is really needed, use Env.getCtx(), as a work-around.